sync/mutex/mutex.go

```diff
@@ -45,3 +56,62 @@ func With2[T, A, B any](m *Mutex[T], f func(*T) (A, B)) (A, B) {
 	})
 	return a, b
 }
+
+// With3 is like With1, but the callback returns three values instead of one.
+func With3[T, A, B, C any](m *Mutex[T], f func(*T) (A, B, C)) (A, B, C) {
```

Comment:
Suggestion:  uses of `Witn3/4` could return a struct with return values?  Seems cleaner.